### PR TITLE
[ABW-3360] Profile claim screen

### PR DESCRIPTION
**Pull request by danvleju-rdx**
Jira ticket: [ABW-3360](https://radixdlt.atlassian.net/browse/ABW-3360)

## Description
Tasks:
- [x] Profile claim screen UI
- [x] Implement functionality in `OverlayWindowClient` to open a full screen
- [ ] Add check logic
- [x] Implement `Clear Wallet on This Phone`
- [ ] Implement `Transfer Control Back to This Phone`

## Screenshot
simulator_screenshot_CA75C630-22A2-498E-A519-8925A5948E26

[ABW-3360]: https://radixdlt.atlassian.net/browse/ABW-3360?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RadixWallet/Features/ClaimWallet/ClaimWallet+Reducer.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	public init() {}
+
+	public func reduce(into state: inout State, viewAction: ViewAction) -> Effect<Action> {
+		switch viewAction {
+		case .clearWalletButtonTapped:
+			.run { send in
+				cacheClient.removeAll()
+				await radixConnectClient.disconnectAll()
+				userDefaults.removeAll()
+				await send(.delegate(.didClearWallet))
```
[the remark this answers is not in the document]
Comment (danvleju-rdx) [reply RadixWallet/Features/ClaimWallet/ClaimWallet+Reducer.swift:39]: Sounds good. This way, we can skip passing the `didResetWallet` delegate action through `FactoryReset` -> `Troubleshooting` -> `Settings` -> `Main` and subscribe to `ResetWalletClient` directly from `Main`

File: RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			case claimWallet(ClaimWallet.State)
+		}
+
+		@CasePathable
+		public enum Action: Sendable, Equatable {
+			case claimWallet(ClaimWallet.Action)
+		}
+
+		public var body: some ReducerOf<Self> {
+			Scope(state: /State.claimWallet, action: /Action.claimWallet) {
```
Comment (kugel3) [inline RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift:33]: Please use the new version of these:
`Scope(state: \.claimWallet, action: \.claimWallet)`

File: RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			}
+		}
+	}
+
+	@Dependency(\.overlayWindowClient) var overlayWindowClient
+
+	public init() {}
+
+	public var body: some ReducerOf<Self> {
+		Scope(state: \.root, action: /Action.child .. ChildAction.root) {
```
[the remark this answers is not in the document]
Comment (danvleju-rdx) [reply RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift:42]: Hmm, not sure about this. Any hint?

[the remark this answers is not in the document]
Comment (kugel3) [reply RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift:42]: `\.child.root`

[the remark this answers is not in the document]
Comment (danvleju-rdx) [reply RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+Reducer.swift:42]: Oh right, forgot about @CasePathable on ChildAction, thx

File: RadixWallet/Features/AppFeature/Overlay/Overlay+View.swift [hunk 4]
```diff
@@ -25,21 +25,30 @@ extension OverlayReducer {
 
 private extension StoreOf<OverlayReducer> {
 	var destination: PresentationStoreOf<OverlayReducer.Destination> {
-		func scopeState(state: State) -> PresentationState<OverlayReducer.Destination.State> {
-			state.$destination
-		}
-		return scope(state: scopeState, action: Action.destination)
+		scope(state: \.$destination, action: \.destination)
```
Comment (kugel3) [inline RadixWallet/Features/AppFeature/Overlay/Overlay+View.swift:28]: This is actually one case where we must use the old version, the compiler goes crazy if we don't

File: RadixWallet/Features/AppFeature/Overlay/Overlay+View.swift [hunk 5]
```diff
@@ -25,21 +25,30 @@ extension OverlayReducer {
...
-			store: destinationStore,
-			state: /OverlayReducer.Destination.State.alert,
-			action: OverlayReducer.Destination.Action.alert
+		return alert(with: destinationStore)
+			.fullScreenCover(with: destinationStore)
+	}
+
+	private func alert(with destinationStore: PresentationStoreOf<OverlayReducer.Destination>) -> some View {
+		alert(
+			store: destinationStore.scope(
```
Comment (kugel3) [inline RadixWallet/Features/AppFeature/Overlay/Overlay+View.swift:42]: This should fit comfortably on a single line

File: RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	@MainActor
+	public struct View: SwiftUI.View {
+		private let store: StoreOf<ClaimWallet>
+
+		public init(store: StoreOf<ClaimWallet>) {
+			self.store = store
+		}
+
+		public var body: some SwiftUI.View {
+			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
```
Comment (kugel3) [inline RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift:28]: And then you can scope to `screenState` here instead

File: RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+			screenState: isLoading ? .loading(.global(text: nil)) : .enabled
+		)
+	}
+}
+
+// MARK: - ClaimWallet.View
+extension ClaimWallet {
+	public struct ViewState: Equatable {
+		let screenState: ControlState
+		// TODO: add ControlState for buttons
```
[the remark this answers is not in the document]
Comment (danvleju-rdx) [reply RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift:16]: A ControlState might be necessary for the profile claiming button, but that is not included in this PR

File: RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+View.swift [hunk 8]
```diff
@@ -13,9 +13,7 @@ extension FullScreenOverlayCoordinator {
 
 		public var body: some SwiftUI.View {
 			NavigationStack {
-				ZStack {
-					root(for: store.scope(state: \.root, action: { .child(.root($0)) }))
-				}
+				root(for: store.scope(state: \.root, action: { .child(.root($0)) }))
```
Comment (kugel3) [inline RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+View.swift:16]: ah, you can probably use `\.child.root` here

File: RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift [hunk 9]
```diff
@@ -52,13 +39,13 @@ extension ClaimWallet {
...
 						Button(L10n.ConfigurationBackup.Automated.walletTransferredClearButton) {
-							viewStore.send(.clearWalletButtonTapped)
+							store.send(.view(.clearWalletButtonTapped))
 						}
 						.buttonStyle(.primaryRectangular)
 						.padding(.bottom, .small2)
 
 						Button(L10n.ConfigurationBackup.Automated.walletTransferredTransferBackButton) {
-							viewStore.send(.transferBackButtonTapped)
+							store.send(.view(.transferBackButtonTapped))
```
Comment (kugel3) [inline RadixWallet/Features/ClaimWallet/ClaimWallet+View.swift:48]: Yeah, might as well stop using viewStore for sending, I also do it like this

**Review by kugel3 — APPROVED** [review]

File: RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+View.swift (hunk 8 above)
Comment (danvleju-rdx) [reply RadixWallet/Features/AppFeature/Overlay/FullScreenOverlayCoordinator+View.swift:16]: right

**Pull request merged**